controller/openebs/reconciler.go

```diff
@@ -108,21 +109,35 @@ func Sync(request *generic.SyncHookRequest, response *generic.SyncHookResponse)
 		hookResponse: response,
 	}
 
-	var openebsObj *unstructured.Unstructured
+	var observedOpenEBS *unstructured.Unstructured
+	var observedOpenEBSComponents []*unstructured.Unstructured
+	// We will use watchNamespace to verify if the various attachments are from the
+	// same namespace or not as of watched resource.
+	watchNamepace := request.Watch.GetNamespace()
 	for _, attachment := range request.Attachments.List() {
 		// this watch resource must be present in the list of attachments
 		if request.Watch.GetUID() == attachment.GetUID() &&
 			attachment.GetKind() == string(types.KindOpenEBS) {
 			// this is the required OpenEBS
-			openebsObj = attachment
+			observedOpenEBS = attachment
 			// add this to the response later after completion of its
 			// reconcile logic
 			continue
 		}
+		if attachment.GetKind() != string(types.KindOpenEBS) {
```

Comment:
We Will have to exempt cluster-level resources from this check or if we add a reference check in the metac.yaml itself, then this check will not be required.